### ISSUE TITLE
E0110 False Positive with nested abstract classes

### ISSUE THREAD

**Issue by pzelnip** [PyCQA/pylint#2344]
### Steps to reproduce

Use the following code:
```
from abc import ABC, abstractmethod


class Message(ABC):
    class Builder(ABC):
        @abstractmethod
        def _instance(self):
            pass

        def __init__(self):
            self._something = None


class SubMessage(Message):
    pass


class ConcreteMessage(SubMessage):
    class Builder(SubMessage.Builder):
        def __init__(self):
            super().__init__()

        def _instance(self):
            return ConcreteMessage()


if __name__ == "__main__":
    ConcreteMessage.Builder()
```

### Current behavior

ConcreteMessage.Builder() is flagged with an E0110 error, even though all abstract methods have been implemented (the fact that the code executes successfully demonstrates this).

### Expected behavior

The ```ConcreteMessage.Builder()``` line should not produce a E0110 error

### pylint --version output

```
No config file found, using default configuration
pylint 1.9.2,
astroid 1.6.4
Python 3.6.2 (default, Jul 17 2017, 16:44:45)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

Edit: I tried the same file in Pylint 2.0.1 and it did *not* produce the same false positive (ie output was as I'd expect in 2.0.1).

**Comment by pzelnip** [comment]
Given that I'm not having this problem in 2.x, I'm totally cool with closing this issue.  If the maintainers feel it's worth fixing in the 1.9.x line then feel free to reopen.
